### PR TITLE
fix: istio destionationrule subsets enforcement

### PR DESCRIPTION
**Pull request by dnguy078**
Checklist:
Relates to issue in https://github.com/argoproj/argo-rollouts/issues/3125 

* [x] Either (a) I've created an [enhancement proposal](https://github.com/argoproj/argo-rollouts/issues/new/choose) and discussed it with the community, (b) this is a bug fix, or (c) this is a chore.
* [x] The title of the PR is (a) [conventional](https://www.conventionalcommits.org/en/v1.0.0/) with a list of types and scopes found [here](https://github.com/argoproj/argo-rollouts/blob/master/.github/workflows/pr-title-check.yml), (b) states what changed, and (c) suffixes the related issues number. E.g. `"fix(controller): Updates such and such. Fixes #1234"`.  
* [x] I've signed my commits with [DCO](https://github.com/argoproj/argoproj)
* [x] I have written unit and/or e2e tests for my change. PRs without these are unlikely to be merged.
* [x] My builds are green. Try syncing with master if they are not. 
* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo-rollouts/blob/master/USERS.md).

### CODE REVIEW

File: rollout/trafficrouting/istio/istio.go [hunk 1]
```diff
@@ -255,6 +255,23 @@ func (r *Reconciler) reconcileVirtualService(obj *unstructured.Unstructured, vsv
 		if err = ValidateHTTPRoutes(r.rollout, vsvcRouteNames, httpRoutes); err != nil {
 			return nil, false, err
 		}
+
+		host, err := r.getDestinationRuleHost()
+		if err != nil {
+			return nil, false, err
+		}
+
+		if host != "" {
```
Comment (agrawroh) [inline rollout/trafficrouting/istio/istio.go:264]: Is it possible to have an empty host? We should check if Istio allows it.

**Review by agrawroh — APPROVED** [review]
Looks good to me overall. @zachaller Could you please take a look as well?

File: rollout/trafficrouting/istio/istio.go (hunk 1 above)
Comment (dnguy078) [reply rollout/trafficrouting/istio/istio.go:264]: looks like the host is required for Istio [DestinationRule](https://istio.io/latest/docs/reference/config/networking/destination-rule/#DestinationRule). We can check do this check on the DestinationRule if included in `.spec.strategy.canary.trafficRouting.istio.destinationRule`

**Comment by dnguy078** [comment]
Thank you for the review @agrawroh !

File: rollout/trafficrouting/istio/istio.go (hunk 1 above)
Comment (dnguy078) [reply rollout/trafficrouting/istio/istio.go:264]: ah that check already exist :) https://github.com/argoproj/argo-rollouts/blob/423be7b9ee37fade38a652b4fc6a43da8c842b5d/rollout/trafficrouting/istio/istio.go#L795

**Review by zachaller — APPROVED** [review]

**Comment by zachaller** [comment]
@agrawroh thanks for the review, I also picked this back to v1.6.1 which I am releasing now

**Comment by zachaller** [comment]
@dnguy078 This might have caused this: https://cloud-native.slack.com/archives/C01U781DW2E/p1698875164861259 you mind taking a peak?

**Comment by zachaller** [comment]
Here is the issue: https://github.com/argoproj/argo-rollouts/issues/3141

**Comment by dnguy078** [comment]
I'll take a look @zachaller

**Pull request merged**